dll/cpl/usrmgr/users.c

```diff
@@ -650,6 +651,50 @@ UpdateUserProperties(HWND hwndDlg)
     NetApiBufferFree(pUserInfo);
 }
 
+void OnToggleRequireLogon(HWND hwndDlg)
+{
+    BOOL bIsChecked;
+    WCHAR szAutoAdminLogonValue[2]; 
+    HKEY hKey;
+    LONG lResult;
+    TCHAR szErrorMsg[256];
+
+    bIsChecked = IsDlgButtonChecked(hwndDlg, IDC_USERS_STARTUP_REQUIRE);
+
+    wcscpy(szAutoAdminLogonValue, (bIsChecked == BST_CHECKED) ? L"0" : L"1");
+
+    lResult = RegCreateKeyExW(HKEY_LOCAL_MACHINE,
+                              L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion\\Winlogon",
+                              0,
+                              NULL,
+                              REG_OPTION_NON_VOLATILE,
+                              KEY_SET_VALUE,
+                              NULL,
+                              &hKey,
+                              NULL);
+
+    if (lResult != ERROR_SUCCESS)
+    {
+        wsprintf(szErrorMsg, L"Failed to open or create Winlogon registry key for writing. Error code: %ld", lResult);
+        MessageBoxW(hwndDlg, szErrorMsg, L"Registry Error", MB_OK | MB_ICONERROR);
+        return; 
+    }
+
+    lResult = RegSetValueExW(hKey,
+                             L"AutoAdminLogon",
+                             0,
+                             REG_SZ, 
+                             (const BYTE*)szAutoAdminLogonValue,
+                             (wcslen(szAutoAdminLogonValue) + 1) * sizeof(WCHAR));
```

Comment:
In this case we know precisely the length of the variable, so:
```suggestion
                             _sizeof(szAutoAdminLogonValue));
```
is in this case fine to use.